### PR TITLE
Wr/sfdx ify

### PR DESCRIPTION
**Pull request by WillieRuemmele**
### What does this PR do?
incorporates `sfdx` behavior/json/config values so that `plugin-settings` can replace `plugin-config` and `plugin-alias`
### What issues does this PR fix or reference?
@W-12083531@

### CODE REVIEW

File: src/commands/alias/set.ts [hunk 1]
```diff
@@ -37,12 +37,16 @@ export default class AliasSet extends AliasCommand<AliasResults> {
...
-          stateAggregator.aliases.set(alias, value);
-          return { alias, success: true, value };
-        }
+        // This is commented out to support plugin-settings in sfdx. Which allows setting an alias to undefined, which
+        // in reality, becomes the string 'undefined'
+        // if (!value) {
+        //   return { alias, success: false, error: messages.createError('error.ValueRequired'), value };
+        // } else {
+        if (!value)
+          this.warn(`Setting an alias to undefined is deprecated v58.0. Use ${this.config.bin} alias unset instead.`);
```
Comment (shetzel) [inline src/commands/alias/set.ts:46]: Should this be, "deprecated as of v57.0."?

**Review by shetzel — APPROVED** [review]

File: src/commands/config/list.ts [hunk 2]
```diff
@@ -14,9 +14,12 @@ export default class List extends ConfigCommand<ConfigResponses> {
   public static readonly description = messages.getMessage('description');
   public static readonly summary = messages.getMessage('summary');
   public static readonly examples = messages.getMessages('examples');
+  public static readonly aliases = ['force:config:list'];
+  public static readonly deprecateAliases = true;
   public static flags = {};
```
Comment (peternhale) [inline src/commands/config/list.ts:19]: Do we need flags?

File: src/commands/config/unset.ts [hunk 3]
```diff
@@ -29,21 +29,42 @@ export class UnSet extends ConfigCommand<ConfigResponses> {
 
   public async run(): Promise<ConfigResponses> {
     const { argv, flags } = await this.parse(UnSet);
+    await SfdxConfigAggregator.create({});
 
     if (!argv || argv.length === 0) {
       throw messages.createError('error.NoConfigKeysFound');
     } else {
```
Comment (peternhale) [inline src/commands/config/unset.ts:36]: Don't need the else

File: src/commands/config/unset.ts [hunk 4]
```diff
@@ -29,21 +29,42 @@ export class UnSet extends ConfigCommand<ConfigResponses> {
...
+              });
+            }
+          } else {
+            this.pushFailure(key, err as Error);
+          }
         }
-      });
+      }
       await config.write();
       if (!this.jsonEnabled()) {
```
Comment (peternhale) [inline src/commands/config/unset.ts:69]: Don't need the guard for !json

File: src/config.ts [hunk 5]
```diff
@@ -35,10 +41,21 @@ export const CONFIG_HELP_SECTION = toHelpSection(
 
 export abstract class ConfigCommand<T> extends SfCommand<T> {
   protected responses: ConfigResponses = [];
-
+  // eslint-disable-next-line class-methods-use-this
+  public calculateSuggestion(cmd: string): string {
```
Comment (peternhale) [inline src/config.ts:45]: Am I mistaken, but this is being used to get suggestions for config keys. Suggest changing `cmd` to something more fitting.

**Review by peternhale — CHANGES_REQUESTED** [review]

File: src/commands/alias/list.ts [hunk 6]
```diff
@@ -16,6 +17,7 @@ export default class AliasList extends AliasCommand<AliasResults> {
   public static examples = messages.getMessages('examples');
   public static readonly aliases = ['force:alias:list'];
   public static readonly deprecateAliases = true;
+  public static readonly flags = { loglevel };
```
Comment (peternhale) [inline src/commands/alias/list.ts:20]: In order to get any warnings regarding loglevel parse must be called

File: src/commands/alias/set.ts [hunk 7]
```diff
@@ -25,6 +25,7 @@ export default class AliasSet extends AliasCommand<AliasResults> {
   public static readonly strict = false; // This allows varargs
   public static readonly aliases = ['force:alias:set'];
   public static readonly deprecateAliases = true;
+  public static readonly flags = { loglevel };
```
Comment (peternhale) [inline src/commands/alias/set.ts:28]: ditto

**Review by peternhale — CHANGES_REQUESTED** [review]

**Comment by shetzel** [comment]
**QA Results**

**Alias:list**
✅  bin/run alias list --help
✅  bin/run alias list
✅  bin/run alias list --json
✅  Compare output with sfdx alias plugin
✅  Aliases work: force alias list, alias list, force:alias:list, alias:list

**Alias:set**
✅  bin/run alias set --help
✅  bin/run alias set foo=bar
✅  bin/run alias set foo="bar bar"
✅  bin/run alias set foo=
✅  bin/run alias set foo bar
✅  bin/run alias set
✅  bin/run alias set foo=bar --json --> has additional entry in result for `"success": true` which is fine
✅  Aliases work: force alias set, alias set, force:alias:set, alias:set

**Alias:unset**
✅  bin/run alias unset --help
✅  bin/run alias unset foo
✅  bin/run alias unset foo bar
✅  bin/run alias unset
✅  bin/run alias unset foo --json  --> has additional entry for `"value": "old value"` and `"success": true` which is fine
✅  bin/run alias set --all
✅  bin/run alias set --all -p
✅  Aliases work: force alias unset, alias unset, force:alias:unset, alias:unset

**Config:list**
✅  bin/run config list --help
✅  bin/run config list
✅  bin/run config list --json  --> has additional entry for `"name": "target-org"` and `"success": true` which is fine
✅  Aliases work: force config list, config list, force:config:list, config:list

**Config:get**
✅  bin/run config get --help
✅  bin/run config get instanceurl  --> "did you mean?"
✅  bin/run config get defaultusername --> warn
✅  bin/run config get target-org
✅  bin/run config get --json  --> has additional entries, all of which are fine
✅  Aliases work: force config get, config get, force:config:get, config:get

**Config:set**
✅  bin/run config set --help
✅  bin/run config set instance-url  --> "did you mean?"
✅  bin/run config set defaultusername --> warn
✅  bin/run config set target-org
❌  bin/run config set org-instance-url --json  --> Different json structure
❌  bin/run config set instanceurl --json  --> incorrect status and success values (should fail)
✅  Aliases work: force config set, config set, force:config:set, config:set


**Config:unset**
✅  bin/run config unset --help
❌  bin/run config unset instance-url  --> "did you mean?" Failed.  It suggested the deprecated key, I selected it and an error was thrown.
✅  bin/run config unset defaultusername --> warn
✅  bin/run config unset target-org
❌  bin/run config unset target-org --json  --> Different json structure
✅  Aliases work: force config unset, config unset, force:config:unset, config:unset

plugin-config results in current latest-rc of sfdx:
Screenshot 2023-02-22 at 3 19 27 PM

plugin-settings results:
Screenshot 2023-02-22 at 3 21 51 PM
Screenshot 2023-02-22 at 3 30 34 PM

**Pull request merged**